### PR TITLE
Nazar/feature/api v4

### PR DESCRIPTION
**Pull request by Nazar-kob**

### CODE REVIEW

File: api/routes/plant.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+def get_all(query_params: s.SearchPlantsQueryParams = Depends(), db: Session = Depends(get_db)):
+    """Returns the plants"""
+    log(log.INFO, "Get plants")
+    query = sa.select(m.PlantVariety)
+    if query_params.name:
+        query = query.where(m.PlantVariety.name.ilike(f"%{query_params.name}%"))
+    if query_params.can_plant_indoors is not None:
+        query = query.where(m.PlantVariety.can_plant_indoors.is_(query_params.can_plant_indoors))
+    if query_params.type_of:
+        query = query.join(m.PlantFamily).where(m.PlantFamily.type_of == query_params.type_of.value)
```
Comment (nixissmtp) [inline api/routes/plant.py:28]: why join?

File: app/models/plant_variety.py [hunk 2]
```diff
@@ -76,6 +76,20 @@ class PlantVariety(db.Model, ModelMixin):
     _photos: orm.Mapped[List["Photo"]] = orm.relationship(secondary=plant_variety_photo)
     programs: orm.Mapped[List["PlantingProgram"]] = orm.relationship("PlantingProgram", back_populates="plant_variety")
 
+    @property
+    def photo(self) -> str | None:  # type:ignore
```
Comment (nixissmtp) [inline app/models/plant_variety.py:91]: WTF is this?

File: app/models/plant_variety.py [hunk 3]
```diff
@@ -76,6 +76,20 @@ class PlantVariety(db.Model, ModelMixin):
     _photos: orm.Mapped[List["Photo"]] = orm.relationship(secondary=plant_variety_photo)
```
Comment (nixissmtp) [inline app/models/plant_variety.py]: make photo write only field

File: app/schema/plant.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    name: str
+    is_sun_loving: bool = Field(alias="isSunLoving")
+    min_size: float = Field(alias="minSize")
+    max_size: float = Field(alias="maxSize")
+
+    model_config = ConfigDict(from_attributes=True, populate_by_name=True, use_enum_values=True)
+
+
+class Plant(BasePlant):
+    photo: Photo | None
```
Comment (nixissmtp) [inline app/schema/plant.py:19]: photos, not photo

File: docker-compose.staging.yaml [hunk 5]
```diff
@@ -15,7 +15,7 @@ services:
       PGUSER: ${POSTGRES_USER:-postgres}
 
   app:
-    image: simple2b/zrosty-hay:0.1.6
+    image: simple2b/zrosty-hay:0.1.7
```
Comment (nixissmtp) [inline docker-compose.staging.yaml:18]: make image version as variable

**Review by nixissmtp — CHANGES_REQUESTED** [review]

File: app/templates/plant_variety/edit_form.html [hunk 6]
```diff
@@ -43,32 +43,19 @@
                     placeholder="Enter watering info...">{{ form.watering_info.data }}</textarea>
             </div>
```
Comment (marynag) [inline app/templates/plant_variety/edit_form.html:34]: Add min and max length for textarea

File: app/templates/plant_variety/edit_form.html [hunk 7]
```diff
@@ -102,6 +89,24 @@
...
+            <div class="flex items-start mb-6">
+                <div class="flex items-center h-5">
+                    <input name="can_plant_indoors"
+                        type="checkbox" "{% if form.can_plant_indoors.data %} checked {% endif %}"
+                        class="w-4 h-4 border border-gray-300 rounded bg-gray-50 focus:ring-3 focus:ring-blue-300 dark:bg-gray-700 dark:border-gray-600 dark:focus:ring-blue-600 dark:ring-offset-gray-800">
+                </div>
+                <label for="can_plant_indoors" class="ms-2 text-sm font-medium text-gray-900 dark:text-gray-300">Can
+                    plant indoors</label>
+            </div>
             <div class="flex items-start mb-6">
```
Comment (marynag) [inline app/templates/plant_variety/edit_form.html:84]: Add required attribute
for adding "*" symbol in input label use:
```
label:has(+ *:required):after,
label:has(~ div *:required):after {
  content: ' *';
  color: red;
}

```

File: docker-compose.staging.yaml (hunk 5 above)
Comment (Nazar-kob) [reply docker-compose.staging.yaml:18]: thanks

**Pull request merged**